src/pip/_internal/commands/install.py

```diff
@@ -284,6 +285,19 @@ def run(self, options: Values, args: List[str]) -> int:
         if options.use_user_site and options.target_dir is not None:
             raise CommandError("Can not combine '--user' and '--target'")
 
+        # Check whether the environment we're installing into is externally
+        # managed, as specified in PEP 668. Specifying --root, --target, or
+        # --prefix disables the check, since there's no reliable way to locate
+        # the EXTERNALLY-MANAGED file for those cases.
+        installing_into_current_environment = (
+            not options.dry_run
```

Comment:
TBH, I would be slightly more comfortable with a `--dry-run` + `--report` skipping the case here. Arguably, a dry run should mirror the behaviour of a real run _except_ for taking any actions and should be performing this check.

Alternatively, a `--stop-after-resolve` name for dry-run as a temporary alias that we migrate to after a migration period might be a better long-term solution here.

Reply:
I added a check for `--report`.